acl/acl_test.go

```diff
@@ -1,930 +1,1400 @@
 package acl
 
 import (
+	"fmt"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
-func TestRootACL(t *testing.T) {
-	if RootACL("allow") != AllowAll() {
-		t.Fatalf("Bad root")
-	}
-	if RootACL("deny") != DenyAll() {
-		t.Fatalf("Bad root")
-	}
-	if RootACL("manage") != ManageAll() {
-		t.Fatalf("Bad root")
-	}
-	if RootACL("foo") != nil {
-		t.Fatalf("bad root")
-	}
+type aclCheck struct {
+	name   string
+	prefix string
+	check  func(t *testing.T, acl ACL, prefix string)
 }
 
-func TestStaticACL(t *testing.T) {
-	all := AllowAll()
-	if _, ok := all.(*StaticACL); !ok {
-		t.Fatalf("expected static")
-	}
+type aclTest struct {
+	name          string
+	defaultPolicy ACL
+	policyStack   []*Policy
+	checks        []aclCheck
+}
```

Comment:
Since this is only used in one test, I recommend just defining it inline in the test function body so its easy to see the structure. Its easy enough to pull it out if we reuse it, but that's fairly common for table tests. You can just do this

```go
func TestFoo(*testing.T) {
  type aclTest struct { ... }

  ...
}
```